xLights/utils/string_utils.h

```diff
@@ -92,6 +93,12 @@ extern const wxString xlEMPTY_WXSTRING;
         return in.find(contains) != std::wstring::npos;
     }
 
+    inline bool hasAlpha(const std::string& str) noexcept 
+    {
+        std::regex pattern("[a-zA-Z]");
+        return std::regex_search(str, pattern);
+    }
+
```

Comment:
Compiling a regex and applying a search for this case is way overkill.    A simple loop would likely perform better:
`
inline bool hasAlpha(const std::string &s) {
    for (auto c : s) {
        if ((c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z')) {
            return true;
        }
    }
    return false;
}
`

Reply:
(https://en.cppreference.com/w/cpp/string/byte/isalpha

Reply:
ha... that was my original implementation, but figured it was ugly....

Reply:
and that too... stsd::isalpha requires the for loop & if...

i can combine both ideas into one..  EVERYONE on board?